addons/account/models/account_analytic_distribution_model.py

```diff
@@ -32,11 +33,34 @@ def _get_default_search_domain_vals(self):
 
     def _get_applicable_models(self, vals):
         applicable_models = super()._get_applicable_models(vals)
-        return applicable_models.filtered(lambda m:
-            not m.account_prefix or vals.get('account_prefix', '').startswith(m.account_prefix)
+
+        # Regex pattern to split by either ';' or ','
+        delimiter_pattern = re.compile(r'[;,]\s*')
+
+        return applicable_models.filtered(lambda model:
+            not model.account_prefix or
+            any((vals.get('account_prefix', '') or '').startswith(prefix) for prefix in delimiter_pattern.split(model.account_prefix))
```

Comment:
had to did (vals.get('account_prefix', '') or '') because there is a traceback if prefix is in vals but false, it will give a False.startswith and crash

happens in master currently

Reply:
Will need to check since when this happened didn't have the TB in 17.0 while I got it in master, but I may have missed it

Reply:
Will check in previous version after 17.0 and make a pr if needed :smile: